src/nevermined/Nfts.ts

```diff
@@ -486,4 +489,132 @@ export class Nfts extends Instantiable {
             from
         )
     }
+
+    public async listOnSecondaryMarkets(
+        agreementId: string,
+        ddo: DDO,
+        assetRewards: AssetRewards,
+        numberNFTs: number,
+        provider: string,
+        owner: Account
+    ): Promise<boolean> {
+        const { nftSalesTemplate } = this.nevermined.keeper.templates
+        const providerAccounts = new Account(provider)
+        const result = await nftSalesTemplate.createAgreementFromDDO(
+            agreementId,
+            ddo,
+            assetRewards,
+            null, // TODO: eliminate needing the consumer
+            numberNFTs,
+            providerAccounts || new Account(this.config.gatewayAddress),
+            owner
+        )
+        if (result) {
+            return true
+        } else {
+            throw Error(`Error listing ${ddo.shortId()} on secondary markets`)
+        }
+    }
+
+    public async buySecondaryMarketNft(
+        buyer: string,
+        seller: string,
+        nftPrice: number,
+        numOfNfts: number = 1,
+        ddo: DDO,
+        agreementId: string
+    ) {
+        const buyersAccount = new Account(buyer)
+        const sellersAccount = new Account(seller)
+        const { token } = this.nevermined.keeper
+
+        //What is config here @Elod
```

Comment:
The question I think is better addressed to @r-marques

Reply:
Same, i didnt do it. @Elod23